src/datamodel.js

```diff
@@ -305,7 +305,13 @@ class DataModel extends Relation {
      * @param {Array.<Array>} sortingDetails - Sorting details based on which the sorting will be performed.
      * @return {DataModel} Returns a new instance of DataModel with sorted data.
      */
-    sort (sortingDetails) {
+    sort (sortingDetails, config = { saveChild: false }) {
+        if (this._sortingDetails.length) {
+            const parent = this._parent;
+            this.dispose();
+            return parent.sort(sortingDetails, config);
+        }
```

Comment:
It breaks the parent-child relationships if the user intention is not to create intermediate datamodels, then he/she should use `compose` operator.
think about the example below:

```javascript
const dm = new DataModel(data, schema);
const dm1 = dm.sort([soring_detials]);

const dm2 = dm1.sort([soring_detials]);
//here, dm1 is destroyed.

callSomeFunction(dm1);
// here user could use it, but dm1 is disposed.
```